### PR TITLE
Adding product version info to WindowsAppRuntime.BootStrap.dll

### PR DESCRIPTION
**Pull request by EricJohnson327**

### CODE REVIEW

File: dev/WindowsAppRuntime_BootstrapDLL/WindowsAppRuntime.Bootstrap.rc [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+ PRODUCTVERSION VER_PRODUCT_VERSION
+ FILEFLAGSMASK 0x3fL
+BEGIN
+    BLOCK "StringFileInfo"
+    BEGIN
+        BLOCK "040904b0"
+        BEGIN
+            VALUE "CompanyName", "Microsoft Corporation"
+            VALUE "FileDescription", "WindowsAppRuntime.Bootstrap.dll"
+            VALUE "InternalName", "WindowsAppRuntime.Bootstrap.dll"
```
Comment (DrusTheAxe) [inline dev/WindowsAppRuntime_BootstrapDLL/WindowsAppRuntime.Bootstrap.rc:22]: `Microsoft.WindowsAppRuntime.Bootstrap.dll`

File: dev/WindowsAppRuntime_BootstrapDLL/WindowsAppRuntime.Bootstrap.rc [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+VS_VERSION_INFO VERSIONINFO
+ PRODUCTVERSION VER_PRODUCT_VERSION
+ FILEFLAGSMASK 0x3fL
+BEGIN
+    BLOCK "StringFileInfo"
+    BEGIN
+        BLOCK "040904b0"
+        BEGIN
+            VALUE "CompanyName", "Microsoft Corporation"
+            VALUE "FileDescription", "WindowsAppRuntime.Bootstrap.dll"
```
Comment (DrusTheAxe) [inline dev/WindowsAppRuntime_BootstrapDLL/WindowsAppRuntime.Bootstrap.rc:21]: `Microsoft.WindowsAppRuntime.Bootstrap.dll`

File: dev/WindowsAppRuntime_BootstrapDLL/WindowsAppRuntime.Bootstrap.rc [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+            VALUE "InternalName", "WindowsAppRuntime.Bootstrap.dll"
+            VALUE "LegalCopyright", "Copyright (c) Microsoft Corporation. All rights reserved."
+            VALUE "OriginalFilename", "WindowsAppRuntime.Bootstrap.dll"
```
Comment (DrusTheAxe) [inline dev/WindowsAppRuntime_BootstrapDLL/WindowsAppRuntime.Bootstrap.rc:24]: `Microsoft.WindowsAppRuntime.Bootstrap.dll`

Comment (DrusTheAxe) [reply dev/WindowsAppRuntime_BootstrapDLL/WindowsAppRuntime.Bootstrap.rc:24]: You can't look at the project filename for the output filename. Not for .vcxproj

Look in the .vcxproj at the `TargetName` element e.g. `<TargetName>Microsoft.WindowsAppRuntime.Bootstrap.dll</TargetName>`

File: dev/WindowsAppRuntime_BootstrapDLL/WindowsAppRuntime.Bootstrap.rc [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+            VALUE "ProductName", "Windows App SDK"
+            VALUE "ProductVersion", VER_PRODUCT_VERSION_STRING "\0"
+        END
+    END
+    BLOCK "VarFileInfo"
+    BEGIN
+        VALUE "Translation", 0x409, 1200
```
Comment (DrusTheAxe) [inline dev/WindowsAppRuntime_BootstrapDLL/WindowsAppRuntime.Bootstrap.rc:31]: If there's no variable info is this BLOCK needed at all?

Remove?

File: dev/WindowsAppRuntime_BootstrapDLL/WindowsAppRuntime.Bootstrap.rc [hunk 5]
```diff
@@ -0,0 +1,33 @@
+#include <winresrc.h>
+
+#define STR1(x) #x
+#define DELIM STR1(.)
+#define STR2(a,b)     STR1(a) DELIM STR1(b)
+
+#define WAS_RELEASE_MAJOR           1
+#define WAS_RELEASE_MINOR           2
```
Comment (DrusTheAxe) [inline dev/WindowsAppRuntime_BootstrapDLL/WindowsAppRuntime.Bootstrap.rc:8]: Don't embed these in the .rc

1. Create `/dev/common/WindowsAppSDK-ProductInfo.h` containing constants

```
// Copyright (c) Microsoft Corporation and Contributors.
// Licensed under the MIT License.

#ifndef __WINDOWSAPPSDK_PRODUCTINFO_H__
#define __WINDOWSAPPSDK_PRODUCTINFO_H__

#ifndef STR2
#define STR1(x) #x
#define VERSION_DELIMIMITER STR1(.)
#define STR2(a,b)     STR1(a) VERSION_DELIMIMITER STR1(b)
#endif

#define WINDOWSAPPSDK_RELEASE_MAJOR                         1
#define WINDOWSAPPSDK_RELEASE_MINOR                         2

#define WINDOWSAPPSDK_PRODUCT_VERSION    WINDOWSAPPSDK_RELEASE_MAJOR, WINDOWSAPPSDK_RELEASE_MINOR, 0, 0
#define WINDOWSAPPSDK_PRODUCT_VERSION_STRING   STR2(WINDOWSAPPSDK_RELEASE_MAJOR, WINDOWSAPPSDK_RELEASE_MINOR)
#define WINDOWSAPPSDK_COMPANY_NAME    "Microsoft Corporation"
#define WINDOWSAPPSDK_LEGAL_COPYRIGHT    "Copyright (c) Microsoft Corporation. All rights reserved."
#define WINDOWSAPPSDK_PRODUCT_NAME    "Windows App SDK"

#endif // __WINDOWSAPPSDK_PRODUCTINFO_H__
```

2. In here line 1.5 `#include <WindowsAppSDK-ProductInfo.h"

That should work as WindowsAppRuntime_BootstrapDLL.vcxproj already has `<AdditionalIncludeDirectories>...$(RepoRoot)\dev\common...`

Comment (DrusTheAxe) [reply dev/WindowsAppRuntime_BootstrapDLL/WindowsAppRuntime.Bootstrap.rc:8]: Then change WindowsAppRuntime.Bootstrap.rc to

```
// Copyright (c) Microsoft Corporation and Contributors.
// Licensed under the MIT License.

#include <winresrc.h>
#include <WindowsAppSDK-ProductInfo.h>

#define VERSIONINFO_FILENAME "Microsoft.WindowsAppRuntime.Bootstrap.dll"

VS_VERSION_INFO VERSIONINFO
 PRODUCTVERSION WINDOWSAPPSDK_PRODUCT_VERSION
 FILEFLAGSMASK 0x3fL
 FILEFLAGS 0
BEGIN
    BLOCK "StringFileInfo"
    BEGIN
        BLOCK "040904b0"
        BEGIN
            VALUE "CompanyName", WINDOWSAPPSDK_COMPANY_NAME
            VALUE "FileDescription", VERSIONINFO_FILENAME
            VALUE "InternalName", VERSIONINFO_FILENAME
            VALUE "LegalCopyright", WINDOWSAPPSDK_LEGAL_COPYRIGHT
            VALUE "OriginalFilename", VERSIONINFO_FILENAME
            VALUE "ProductName", WINDOWSAPPSDK_PRODUCT_NAME
            VALUE "ProductVersion", WINDOWSAPPSDK_PRODUCT_VERSION_STRING "\0"
        END
    END
END
```

Comment (DrusTheAxe) [reply dev/WindowsAppRuntime_BootstrapDLL/WindowsAppRuntime.Bootstrap.rc:8]: Then you also have a template for everything else in the repo

1. If .rc doesn't exist add a copy of this; If .rc already exists add the #include WindowsAppSDK-ProductInfo.h on down
2. Change the VERSIONINFO_FILENAME value

And then in future releases you only need to change the major+minor constants in the .h (2 values in 1 file) and magic happens

**Review by DrusTheAxe — CHANGES_REQUESTED** [review]

File: dev/WindowsAppRuntime_BootstrapDLL/WindowsAppRuntime.Bootstrap.rc (hunk 5 above)
Comment (EricJohnson327) [reply dev/WindowsAppRuntime_BootstrapDLL/WindowsAppRuntime.Bootstrap.rc]: Thanks!  Updated

File: dev/WindowsAppRuntime_BootstrapDLL/WindowsAppRuntime.Bootstrap.rc (hunk 4 above)
Comment (EricJohnson327) [reply dev/WindowsAppRuntime_BootstrapDLL/WindowsAppRuntime.Bootstrap.rc:29]: Unfortunately yes.  The translation fields says what language to make the file info.  Without it, nothing shows up.

**Review by DrusTheAxe — APPROVED** [review]

**Pull request merged**